nestkernel/universal_data_logger.h

```diff
@@ -241,13 +243,12 @@ nest::UniversalDataLogger< HostNode >::connect_logging_device( const DataLogging
 
   // ensure that we have not connected this multimeter before
   const index mm_node_id = req.get_sender().get_node_id();
-  const size_t n_loggers = data_loggers_.size();
-  size_t j = 0;
-  while ( j < n_loggers and data_loggers_[ j ].get_mm_node_id() != mm_node_id )
-  {
-    ++j;
-  }
-  if ( j < n_loggers )
+
+  auto item = std::find_if( data_loggers_.begin(),
+    data_loggers_.end(),
+    [ &mm_node_id ]( const DataLogger_& dl ) { return dl.get_mm_node_id() == mm_node_id; } );
```

Comment:
I think there's no need to repeat the name here. `[ & ]` captures all variables used in the lambda by reference.

```suggestion
    [ & ]( const DataLogger_& dl ) { return dl.get_mm_node_id() == mm_node_id; } );
```